### PR TITLE
Ankit | Test | UI implementation

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cv4gcnc

### CODE REVIEW

File: apps/web-giddh/src/app/services/subscriptions.service.ts [hunk 1]
```diff
@@ -49,6 +49,14 @@ export class SubscriptionsService {
                 return data;
             }), catchError((e) => this.errorHandler.HandleCatch<string, string>(e, '')));
     }
+    public saveRazorpayToken(subscriptionId: string , paymentId: string): Observable<any> {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/subscriptions.service.ts:60]: Js Doc

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/services/subscriptions.service.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/services/subscriptions.service.ts:60]: done

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 2]
```diff
@@ -1420,7 +1453,11 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
             };
             this.subscriptionId = subscription?.subscriptionId;
             let data = { ...request, ...this.subscriptionRequest };
-            this.componentStore.changePlan(data);
+            if(subscription?.duration === 'MONTHLY' && subscription?.region?.code !== 'GBR') {
+               this.componentStore.saveRazorpayToken({subscriptionId:this.subscriptionId, paymentId: !zeroAmount ? payResponse.razorpay_payment_id : null});
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]:
```suggestion
               this.componentStore.saveRazorpayToken({subscriptionId:this.subscriptionId, request?.paymentId});
```

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 3]
```diff
@@ -392,6 +395,44 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
...
+    readonly saveRazorpayToken  = this.effect((data: Observable<any>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({ razorpaySuccess: null });
+                return this.subscriptionService.saveRazorpayToken(req.subscriptionId,req.paymentId).pipe(
+                    tapResponse(
+                        (res: any) => {
+                            if (res?.status === 'success') {
+                                return this.patchState({
+                                    razorpaySuccess: true,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]:
```suggestion
                                    razorpaySuccess: true
```

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 4]
```diff
@@ -392,6 +395,44 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
...
+                                });
+                            } else {
+                                if (res.message) {
+                                    this.toasterService.showSnackBar('error', res.message);
+                                }
+                                return this.patchState({
+                                    razorpaySuccess: false,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]:
```suggestion
                                    razorpaySuccess: false
```

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 5]
```diff
@@ -392,6 +395,44 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
...
+                                });
+                            }
+                        },
+                        (error: any) => {
+                            this.toasterService.showSnackBar('error', 'Something went wrong! Please try again.');
+                            return this.patchState({
+                                razorpaySuccess: false,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]:
```suggestion
                                razorpaySuccess: false
```

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 6]
```diff
@@ -71,6 +73,7 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
     public onboardingForm$: Observable<any> = this.select(this.store.select(state => state.common.onboardingform), (response) => response);
     public commonCountries$: Observable<any> = this.select(this.store.select(state => state.common.countries), (response) => response);
     public generalState$: Observable<any> = this.select(this.store.select(state => state.general.states), (response) => response);
+    public razorpaySuccess$ = this.select((state) => state.razorpaySuccess);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: move into component

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]: done

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: done

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: done

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: done

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**